### ISSUE TITLE
[java] MissingOverride is triggered in illegal places

### ISSUE THREAD

**Issue by huxi** [pmd/pmd#1063]
**Affects PMD Version:** 
6.3.0

**Rule:**
`<rule ref="category/java/bestpractices.xml/MissingOverride" />`

**Description:**
`MissingOverride` is triggered in places where adding `@Override` would be an error.

**Code Sample demonstrating the issue:**

I found two such cases.

An interface extending another interface to specialize the return type of a method.
```
interface Foo extends Cloneable {
    // triggers MissingOverride but mustn't
    Foo clone() throws CloneNotSupportedException;
}
```

Static methods that happen to have the same name in a class hierarchy.
```
public class FooBase {
    public static FooBase parse(String input) {
        // ...
    }
}

public class FooA extends FooBase {
    // triggers MissingOverride but mustn't
    public static FooA parse(String input) {
        // ...
    }
}

public class FooB extends FooBase {
    // triggers MissingOverride but mustn't
    public static FooB parse(String input) {
        // ...
    }
}
```

**Running PMD through:** *Gradle*

**Comment by boris-petrov** [comment]
For me it also triggers for both:

```java
private void writeObject(@SuppressWarnings("unused") ObjectOutputStream out) throws NotSerializableException {
	throw new NotSerializableException();
}

private void readObject(@SuppressWarnings("unused") ObjectInputStream in) throws NotSerializableException {
	throw new NotSerializableException();
}
```

**Comment by mattnelson** [comment]
Found an additional scenario, if you want another test case.

```java
class One {
  public static void main(final String[] args) {
    // do stuff
  }
}

class Two extends One {
  // triggers MissingOverride
  public static void main(final String[] args) {
    // do stuff
  }
}
```

**Comment by oowekyala** [comment]
Thanks everyone for your input! We weren't filtering out static or private methods, this will be fixed shortly
